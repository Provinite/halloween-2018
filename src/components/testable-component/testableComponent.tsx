import * as React from "react";

/**
 * A component with a simple derived state.
 */
export class TestableComponent extends React.Component<
  {
    leftSummand: number;
    rightSummand: number;
  },
  {
    sum: number;
  }
> {
  static getDerivedStateFromProps(props, state) {
    if (state.sum !== TestableComponent.add(props)) {
      return {
        sum: TestableComponent.add(props)
      };
    }
    return null;
  }

  private static add(props): number {
    return props.leftSummand + props.rightSummand;
  }
  constructor(props) {
    super(props);
    this.state = {
      sum: TestableComponent.add(props)
    };
  }

  render() {
    return (
      <span>
        The result of {this.props.leftSummand} + {this.props.rightSummand} is
        {this.state.sum}
      </span>
    );
  }
}
