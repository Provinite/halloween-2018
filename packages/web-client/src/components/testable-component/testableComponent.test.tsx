import * as React from "react";
import * as renderer from "react-test-renderer";
import { TestableComponent } from "./testableComponent";
test("state", () => {
  const component = renderer.create(
    <TestableComponent leftSummand={5} rightSummand={7} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
