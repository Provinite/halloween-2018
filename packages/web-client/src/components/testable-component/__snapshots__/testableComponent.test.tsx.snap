// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state 1`] = `
<span>
  The result of 
  5
   + 
  7
   is
  12
</span>
`;
